src/seedu/addressbook/data/AddressBook.java

```diff
@@ -128,4 +129,34 @@ public boolean equals(Object other) {
                         && this.allPersons.equals(((AddressBook) other).allPersons)
                         && this.allTags.equals(((AddressBook) other).allTags));
     }
+
+    /**
+     * Checks if original tag exists in master tag list
+     * and carry out replacement of the tag for all persons in addressbook.
+     *
+     * @param targetTag is the original tag to be replaced.
+     * @param newTag is the new tag to replace the original.
+     * @return true if original tag exists and all such tags have been replaced. Returns false if original tag does not
+     * exists.
+     */
+    public boolean checkAndReplaceTag(Tag targetTag, Tag newTag){
+        if (!allTags.contains(targetTag)) {
+            return false;
+        }
+        else {
+            boolean newTagExists = allTags.contains(newTag);
+
+            for (Person p: allPersons) {
+                boolean hasTagsChanged = p.checkPersonTagList(targetTag, newTag);
+                if (!newTagExists && hasTagsChanged) {
+                    syncTagsWithMasterList(p);
+                    newTagExists = true;
+                }
+            }
+
+
```

Comment:
Use blank lines sparingly. It's good to have blank lines to separate logical code blocks, but too many blank lines will increase the height of the code, which increases the amount of scrolling.